common/src/main/kotlin/org/bibletranslationtools/trchunkbrowser/common/domain/FileProcessor.kt

```diff
@@ -0,0 +1,8 @@
+package org.bibletranslationtools.trchunkbrowser.common.domain
+
+import org.bibletranslationtools.trchunkbrowser.common.model.audio.BttrWavFile
+import java.io.File
+
+interface FileProcessor {
+    fun process(file: File): BttrWavFile
```

Comment:
I see a design pattern (Chain of Responsibility) used here. Returning a "product" after processing is a good choice. However, this may not capture the case when your processor failed and it would not always be able to produce a `BttrWaveFile`.


![image](https://github.com/WycliffeAssociates/tr-chunk-browser/assets/34975907/e0fea3a4-f04d-43ba-ac93-d0ec230a54c1)

 
![image](https://github.com/WycliffeAssociates/tr-chunk-browser/assets/34975907/00b4e25f-2c84-4846-a163-90afd3dfe4dd)

Reply:
That's a good point. Though the processors are executed in GetWavSegments.segments observable and when an Exception occurred, it will return empty list. I just didn't want to fix the old code.